### PR TITLE
chore(new_relic_logs sink): Use `Transformer` in `new_relic_logs` sink and remove `EncodingConfigWithDefault`

### PR DESCRIPTION
**Pull request by pablosichert**
Related to #9459.

While we don't allow setting a codec (just as previously), we deprecate the usage of the former `EncodingConfigWithDefault` for the new `Transformer` here, for internal consistency and to make the intent clear.

Technically, this is a breaking change because setting

```toml
encoding = "json"
```

or

```toml
encoding.codec = "json"
```

will be implied and will be rejected by the config. However, this behavior has not been documented before. Therefore we don't mark it as breaking.

### CODE REVIEW

**Comment by pablosichert** [comment]
@jszwedko you were too quick, I made an edit to the PR description 😄

**Comment by jszwedko** [comment]
Ah, interesting, does that affect all sinks that use `Transformer`? Or what is the difference with `new_relic_logs`?

**Comment by pablosichert** [comment]
@jszwedko `new_relic` uses `EncodingConfigFixed` (which does not provide a user configurable codec), `new_relic_logs` here uses `EncodingConfigWithDefault`.

The other ones that used `EncodingConfigWithDefault` might be a breaking change as well. However, some of them defined a bogus enum like `Encoding { Default }`, and I'm assuming nobody actually explicitly configured


```toml
encoding = "default"
```

or

```toml
encoding.codec = "default"
```

**Comment by jszwedko** [comment]
Ah, gotcha, I see. Since we didn't document that `encoding.codec` could be set on the `new_relic_logs` sink I think we can avoid marking this as a breaking change.

File: src/sinks/new_relic_logs.rs [hunk 1]
```diff
@@ -79,23 +79,7 @@ inventory::submit! {
...
-    #[derivative(Default)]
-    Json,
-}
-
-impl From<Encoding> for crate::sinks::http::Encoding {
-    fn from(v: Encoding) -> crate::sinks::http::Encoding {
-        match v {
-            Encoding::Json => crate::sinks::http::Encoding::Json,
-        }
+        toml::Value::try_from(Self::default()).unwrap()
```
Comment (jszwedko) [inline src/sinks/new_relic_logs.rs:82]: Why is the `unwrap` safe here? I might make it `expect(...)`.

**Review by jszwedko — APPROVED** [review]
One comment, otherwise 👍

File: src/sinks/new_relic_logs.rs (hunk 1 above)
Comment (jszwedko) [reply src/sinks/new_relic_logs.rs:82]: Aha 👍 Fair enough. It just wasn't immediately clear why this was safe. I still think we should generally prefer `expect()` over `unwrap()` so we can document why it is safe.

**Review by spencergilbert — APPROVED** [review]
Noting that this sink is deprecated

**Review by lukesteensen — APPROVED** [review]

**Pull request merged**